pkg/cmd/cli/cmd/set/imagelookup.go

```diff
@@ -0,0 +1,235 @@
+package set
+
+import (
+	"fmt"
+	"io"
+	"text/tabwriter"
+
+	"github.com/golang/glog"
+	"github.com/spf13/cobra"
+
+	kapi "k8s.io/kubernetes/pkg/api"
+	"k8s.io/kubernetes/pkg/api/meta"
+	"k8s.io/kubernetes/pkg/api/unversioned"
+	kcmdutil "k8s.io/kubernetes/pkg/kubectl/cmd/util"
+	"k8s.io/kubernetes/pkg/kubectl/resource"
+	"k8s.io/kubernetes/pkg/runtime"
+
+	"github.com/openshift/origin/pkg/cmd/templates"
+	"github.com/openshift/origin/pkg/cmd/util/clientcmd"
+	imageapi "github.com/openshift/origin/pkg/image/api"
+)
+
+var (
+	imageLookupLong = templates.LongDesc(`
+		Use an image stream from pods and other objects
+
+		Image streams make it easy to tag images, track changes from other registries, and centralize
+		access control to images. Local name lookup allows an image stream to be the source of
+		images for pods, deployments, replica sets, and other resources that reference images, without
+		having to provide the full registry URL. If	local name lookup is enabled for an image stream
+		named 'mysql', a pod or other resource that references 'mysql:latest' (or any other tag) will
+		pull from the location specified by the image stream tag, not from an upstream registry.
+
+		Once lookup is enabled, simply reference the image stream tag in the image field of the object.
+		For example:
+
+		    $ oc import-image mysql:latest --confirm
+				$ %[1]s image-lookup mysql
+				$ oc run mysql --image=mysql
+
+		will import the latest MySQL image from the DockerHub, set that image stream to handle the
+		"mysql" name within the project, and then launch a deployment that points to the image we
+		imported.`)
+
+	imageLookupExample = templates.Examples(`
+		# Print all of the image streams and whether they resolve local names
+	  %[1]s image-lookup
+
+	  # Use local name lookup on image stream mysql
+	  %[1]s image-lookup mysql
+
+	  # Disable local name lookup on image stream mysql
+	  %[1]s image-lookup mysql --enabled=false
+
+	  # Set local name lookup on all image streams
+	  %[1]s image-lookup --all`)
+)
+
+type ImageLookupOptions struct {
+	Out io.Writer
+	Err io.Writer
+
+	Filenames []string
+	Selector  string
+	All       bool
+
+	Builder *resource.Builder
+	Infos   []*resource.Info
+
+	Encoder runtime.Encoder
+
+	ShortOutput   bool
+	Mapper        meta.RESTMapper
+	OutputVersion unversioned.GroupVersion
+
+	PrintTable  bool
+	PrintObject func(runtime.Object) error
+
+	Enabled bool
+}
+
+// NewCmdImageLookup implements the set image-lookup command
+func NewCmdImageLookup(fullName string, f *clientcmd.Factory, out, errOut io.Writer) *cobra.Command {
+	options := &ImageLookupOptions{
+		Out:     out,
+		Err:     errOut,
+		Enabled: true,
+	}
+	cmd := &cobra.Command{
+		Use:     "image-lookup STREAMNAME [...]",
```

Comment:
Another way to say this is that we're allowing an image stream to "own" lookups for its name.  If we don't want to use `Resolve` on the API field we could potentially try:

* `ReplaceLocalNames`
* `LocalImageLookup`
* `LocalNameLookup`
* `LocalLookup`
* `LookupByName`

I don't know that I love any of those - I feel like we're missing a really good verb.